packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -412,19 +419,35 @@ export async function transformGlobImport(
 
           const resolvePaths = (file: string) => {
             if (!dir) {
-              if (isRelative)
+              if (!options.base && isRelative)
```

Comment:
With the updates only to `dir` in my comments, I believe we can then revert this part, from line 422 to 450, as I think it should automatically work then.